addons/mail/static/src/discuss/discuss.xml

```diff
@@ -10,9 +10,21 @@
             <div class="px-3 bg-100 border-bottom d-flex flex-shrink-0" style="height:42px;">
                 <ThreadIcon className="'me-2 align-self-center'" thread="thread"/>
                 <div class="d-flex flex-grow-1 align-self-center align-items-center">
-                    <div class="lead flex-shrink-0 px-1 border text-truncate fw-bold border-light" t-att-title="thread.name" >
-                        <t t-esc="thread.name"/>
-                    </div>
+                    <t t-if="!state.isEditingThreadName">
+                        <div class="lead flex-shrink-0 px-1 border text-truncate fw-bold" t-att-title="thread.name" t-on-click="onClickThreadName" t-on-mouseenter="onMouseEnterThreadName" t-on-mouseleave="onMouseLeaveThreadName" t-attf-class="{{ state.isMouseOverThreadName ? 'o-threadNameEditable bg-white' : 'border-light' }}">
+                            <t t-esc="thread.name"/>
+                        </div>
+                    </t>
+                    <t t-if="state.isEditingThreadName">
+                        <input
+                            class="o_ThreadViewTopbar_threadNameInput lead fw-bold"
+                            t-att-value="state.pendingThreadName"
+                            t-on-input="onInputThreadNameInput"
+                            t-on-keydown="onKeyDownThreadNameInput"
+                            t-ref="threadNameInput"
```

Comment:
Most of it can just become `t-model`, so that content of `input` is sync'ed with a local state on component.
When pressing `ENTER`, should trim and perform the rename.